pulp_ansible/tests/functional/conftest.py

```diff
@@ -181,10 +202,13 @@ def _ansible_git_remote_factory(**kwargs):
 def build_and_upload_collection(ansible_collection_version_api_client):
     """A factory to locally create, build, and upload a collection."""
 
-    def _build_and_upload_collection():
-        collection = build_collection("skeleton")
-        response = ansible_collection_version_api_client.create(file=collection.filename)
+    def _build_and_upload_collection(ansible_repo=None, **kwargs):
+        collection = build_collection("skeleton", **kwargs)
+        body = {"file": collection.filename}
+        if ansible_repo:
+            body["repository"] = ansible_repo.pulp_href
+        response = ansible_collection_version_api_client.create(**body)
         task = monitor_task(response.task)
-        return collection, task.created_resources[0]
+        return collection, task.created_resources[-1]
```

Comment:
I don't think this is safe. As far as I know there is no guarantee on the order of resources.